tests/gui/test_widgets.py

```diff
@@ -40,41 +42,40 @@ def test_ToolBar_setup(mocker):
 
 
 def test_ToolBar_set_logged_in_as(mocker):
-    """Given a username, the user_state is updated and login/logout
-    buttons, and refresh buttons, are in the correct state.
+    """
+    When a user is logged in check that buttons and menus are in the correct state.
     """
     tb = ToolBar(None)
-
-    tb.user_state = mocker.MagicMock()
     tb.login = mocker.MagicMock()
-    tb.logout = mocker.MagicMock()
-    tb.refresh = mocker.MagicMock()
+    tb.user_icon = mocker.MagicMock()
+    tb.user_state = mocker.MagicMock()
+    tb.user_menu = mocker.MagicMock()
 
     tb.set_logged_in_as('test')
 
     tb.user_state.setText.assert_called_once_with('test')
-    tb.login.setVisible.assert_called_once_with(False)
-    tb.logout.setVisible.assert_called_once_with(True)
-    tb.refresh.setVisible.assert_called_once_with(True)
+    tb.login.hide.assert_called_once_with()
+    tb.user_icon.show.assert_called_once_with()
+    tb.user_state.show.assert_called_once_with()
+    tb.user_menu.show.assert_called_once_with()
 
 
 def test_ToolBar_set_logged_out(mocker):
     """
-    Ensure the UI reverts to the logged out state.
+    When a user is logged out check that buttons and menus are in the correct state.
     """
     tb = ToolBar(None)
-
-    tb.user_state = mocker.MagicMock()
     tb.login = mocker.MagicMock()
-    tb.logout = mocker.MagicMock()
-    tb.refresh = mocker.MagicMock()
+    tb.user_icon = mocker.MagicMock()
+    tb.user_state = mocker.MagicMock()
+    tb.user_menu = mocker.MagicMock()
 
     tb.set_logged_out()
 
-    tb.user_state.setText.assert_called_once_with('Signed out.')
-    tb.login.setVisible.assert_called_once_with(True)
-    tb.logout.setVisible.assert_called_once_with(False)
-    tb.refresh.setVisible.assert_called_once_with(False)
```

Comment:
(and we should keep this assert in the test once we hide the refresh button after logout)

Reply:
Same response as my other comment about the upcoming PR. But now that I'm thinking about this more, I can switch the behavior back so this is a completely tested PR, and then just update the tests later.